x-pack/plugins/cases/server/client/cases/validators.ts

```diff
@@ -133,3 +137,49 @@ export const validateRequiredCustomFields = ({
     );
   }
 };
+
+export const validateSearchCasesCustomFields = ({
+  customFieldsConfiguration,
+  customFields,
+}: {
+  customFieldsConfiguration: CustomFieldsConfiguration;
+  customFields: CasesSearchRequest['customFields'];
+}) => {
+  let customFieldsMapping: ICasesCustomField | null | undefined = null;
+
+  if (!customFields || isEmpty(customFields)) {
+    return;
+  }
+
+  if (!customFieldsConfiguration.length) {
+    throw Boom.badRequest('No custom fields configured.');
+  }
+
+  if (Object.keys(customFields).length > MAX_CUSTOM_FIELDS_PER_CASE) {
+    throw Boom.forbidden(`Maximum ${MAX_CUSTOM_FIELDS_PER_CASE} customFields are allowed.`);
+  }
+
+  Object.entries(customFields).forEach(([key, value]) => {
+    const customFieldConfig = customFieldsConfiguration.find((config) => config.key === key);
+
+    if (customFieldConfig) {
+      customFieldsMapping = casesCustomFields.get(customFieldConfig.type);
+
+      if (!customFieldsMapping?.isFilterable) {
+        throw Boom.forbidden(
+          `Filtering by custom field of type ${customFieldConfig.type} is not allowed.`
+        );
+      }
+
+      value.forEach((item) => {
```

Comment:
What if value is `[]`? Should we change the type in `x-pack/plugins/cases/common/types/api/case/v1.ts` to a `limitedArraySchema`?

Reply:
Right now if value is `[]`, it returns all cases. Maybe we should throw error to have at least 1 value.
I didn't add `limitedArraySchema ` because currently we don't know what should be the max limit. We can update it when we implement list as customField.